src/shared/telemetry/defaultTelemetryService.ts

```diff
@@ -271,28 +278,22 @@ export class DefaultTelemetryService implements TelemetryService {
     }
 
     /**
-     * Only passive telemetry is allowed during startup (except for some known
+     * Verifies that telemetry queue only has passive metrics (except for some known
      * special-cases).
+     *
+     * Used while validating extension activation.
      */
     public assertPassiveTelemetry(didReload: boolean) {
         // Special case: these may be non-passive during a VSCode "reload". #1592
         const maybeActiveOnReload = ['sam_init']
-        // Metrics from the previous session can be arbitrary: we can't reason
-        // about whether they should be passive/active.
-        let readingCache = true
 
         // Array for..of is in-order:
         // https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/forEach
         for (const metric of this._eventQueue) {
-            if (readingCache) {
-                readingCache = metric !== this._endOfCache
-                // Skip cached metrics.
-                continue
-            }
             if (metric.Passive || (didReload && maybeActiveOnReload.includes(metric.MetricName))) {
                 continue
             }
-            const msg = `non-passive metric emitted at startup: ${metric.MetricName}`
+            const msg = `non-passive metric detected in queue: ${metric.MetricName}`
```

Comment:
This change is questionable. The important part of the message is the time (during startup).

Reply:
Nothing enforces that this function is called at startup though. Removing the old cache implementation removes that tie. I've addressed this in the function's documentation, but we could even change the function's name (or set a boolean so this can't get called again).

Reply:
> or set a boolean so this can't get called again

🤔 seems appropriate here